servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -931,18 +1022,57 @@ public boolean rename(long fileId, TachyonURI dstPath)
 
       // Now we remove srcInode from it's parent and insert it into dstPath's parent
       long opTimeMs = System.currentTimeMillis();
-      renameInternal(fileId, dstPath, opTimeMs);
+      if (!renameInternal(fileId, dstPath, true, opTimeMs)) {
+        return false;
+      }
 
       writeJournalEntry(new RenameEntry(fileId, dstPath.getPath(), opTimeMs));
       flushJournal();
 
+      LOG.debug("Renamed " + srcPath + " to " + dstPath);
       return true;
     }
   }
 
-  void renameInternal(long fileId, TachyonURI dstPath, long opTimeMs)
-      throws InvalidPathException, FileDoesNotExistException {
+  /**
+   * Implements renaming.
+   *
+   * @param fileId the file id of the rename source
+   * @param dstPath the path to the rename destionation
+   * @param persisted whether the operation should be persisted to UFS
+   * @param opTimeMs the time of the operation
+   * @return
+   * @throws FileDoesNotExistException if a non-existent file is encountered
+   * @throws InvalidPathException if an invalid path is encountered
+   * @throws IOException if an I/O error is encountered
+   */
+  boolean renameInternal(long fileId, TachyonURI dstPath, boolean persisted, long opTimeMs)
+      throws FileDoesNotExistException, InvalidPathException, IOException {
+    // This function should only be called from within synchronized (mInodeTree) blocks.
     Inode srcInode = mInodeTree.getInodeById(fileId);
+    TachyonURI srcPath = mInodeTree.getPath(srcInode);
+    LOG.debug("Renaming " + srcPath + " to " + dstPath);
+
+    // If the source file is persisted, rename it in the UFS.
+    FileInfo fileInfo = getFileInfoInternal(srcInode);
+    if (persisted && fileInfo.isPersisted) {
```

Comment:
The meaning of the two persists were different, which was confusing. I replaced ``persisted`` with ``replayed``, which indicates whether we are in journal replay mode or not.